nntrainer/tensor/manager.cpp

```diff
@@ -172,24 +174,15 @@ void Manager::trackWeights(std::vector<Weight> &ws) {
   total_weight_size += weight_size;
   total_grad_size += grad_size;
   max_grad_size = std::max(max_grad_size, grad_size);
+  weights_initialized = false;
 }
 
-/**
- * @brief Allocate and initialize the weight variable
- */
-void Manager::initialize() {
-  if (total_weight_size == 0) {
-    ml_logw("Nothing done on initialize because there is no weight registered");
-    return;
-  }
-  using AllocFunc = std::function<Tensor(const TensorDim &, size_t)>;
-
+Manager::AllocFunc Manager::getAllocFunc(bool is_weight) {
   AllocFunc allocate_none = [](const TensorDim &dim, size_t) {
     return Tensor();
   };
 
-  AllocFunc allocate_weight = allocate_none;
-  AllocFunc allocate_grad = allocate_none;
+  AllocFunc allocate_func = allocate_none;
```

Comment:
I wonder that if the user_shared_memory is false, then, the current implementation is working?

Reply:
`unittest_nntrainer_models` runs all its unit tests twice - once with all the optimizations on and once again with all the optimizations off. So, I think it works.
Currently, we don't check different combinations of optimizations which IMO would be too much.